File: server/models/document.js
const { Schema, model } = require("mongoose");

const documentSchema = Schema({
  domainId: { type: Schema.Types.ObjectId, required: true, ref: "Domain" },
  title: { type: String, required: true, trim: true },
  text: { type: String, required: true, trim: true },
  tasks: [{ type: Schema.Types.ObjectId, ref: "Task" }],
  status: { type: String, required: true },
  dtStatus: { type: Date, default: Date.now() },
  dtCr: { type: Date, default: Date.now() },
  dtUp: Date,
});

documentSchema.index({ title: "text", text: "text" });

documentSchema.set("toObject", { virtuals: true });
documentSchema.set("toJson", { virtuals: true });

exports.Document = model("Document", documentSchema);

File: marina/lib/common/models/document.dart
import 'package:marina/common/models/domain.dart';

class Document {
  String? id;
  Domain? domainId;
  String? title;
  String? text;

  Document({this.id, this.domainId, this.title, this.text});

  factory Document.fromJson(Map<String, dynamic> json) => Document(
    id: json["_id"],
    domainId: Domain.fromJson(json["domainId"]),
    title: json["title"],
    text: json["text"],
  );

  Map<String, dynamic> toJson() => {
    "_id": id,
    "domainId": domainId!.toJson(),
    "title": title,
    "text": text,
  };
}
